tests/components/media_player/test_samsungtv.py

```diff
@@ -123,22 +131,46 @@ def test_send_key(self):
     def test_send_key_broken_pipe(self):
         """Testing broken pipe Exception."""
         _remote = mock.Mock()
-        self.device.get_remote = mock.Mock()
         _remote.control = mock.Mock(
-            side_effect=BrokenPipeError("Boom"))
-        self.device.get_remote.return_value = _remote
-        self.device.send_key("HELLO")
+            side_effect=BrokenPipeError('Boom'))
+        self.device.get_remote = mock.Mock(return_value=_remote)
+        self.device.send_key('HELLO')
+        self.assertIsNone(self.device._remote)
+        self.assertEqual(STATE_ON, self.device._state)
+
+    def test_send_key_connection_closed_retry_succeed(self):
+        """Test retry on connection closed."""
+        _remote = mock.Mock()
+        _remote.control = mock.Mock(side_effect=[
+            self.device._exceptions_class.ConnectionClosed('Boom'),
+            mock.DEFAULT])
+        self.device.get_remote = mock.Mock(return_value=_remote)
+        command = 'HELLO'
+        self.device.send_key(command)
+        self.assertEqual(STATE_ON, self.device._state)
+        # verify that _remote.control() get called twice because of retry logic
+        _remote.assert_has_calls([
```

Comment:
We prefer to use `assert` statement and `mock.call_args`, to avoid a test passing despite error if `assert_has_calls` is misspelled.

Reply:
Good to know. I updated my PR with the suggested change.